### PR TITLE
Support for marking devices as verified

### PR DESCRIPTION
**Pull request by richvdh**
Add device info to the MemberInfo, and to the User settings, and add a way to
mark other devices as verified.

Check if the device is verified when displaying encrypted events, and add a CSS
class accordingly.

Requires https://github.com/matrix-org/matrix-js-sdk/pull/136

### CODE REVIEW

File: src/components/views/rooms/EventTile.js [hunk 1]
```diff
@@ -152,6 +160,32 @@ module.exports = React.createClass({
         return false;
     },
 
+    componentWillUnmount: function() {
+        dispatcher.unregister(this.dispatcherRef);
+    },
+
+    onAction: function(payload) {
+        switch (payload.action) {
+            case 'device_verified':
```
Comment (dbkr) [inline src/components/views/rooms/EventTile.js]: Is the ultimate plan to store device verifications on the home server & sync them between devices? If so, presumably this should be an event that comes out of the client so a client can update if you verify someone's device from another of your devices.

File: src/components/views/rooms/MemberDeviceInfo.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                userId: this.props.userId,
+                deviceId: this.props.device.id,
+            },
+        });
+    },
+
+    render: function() {
+        var indicator = null, button = null;
+        if (this.props.device.verified) {
+            indicator = (
```
Comment (dbkr) [inline src/components/views/rooms/MemberDeviceInfo.js]: Should there be a way to un-verify a device?

**Comment by dbkr** [comment]
Couple of comments, but all things that can come later: as a first version, lgtm.

File: src/components/views/rooms/EventTile.js (hunk 1 above)
Comment (richvdh) [inline src/components/views/rooms/EventTile.js]: Hum; I hadn't thought about that. The problem is that a naive approach means that the HS could fiddle with the list of devices you trust. I suppose we need this, but there would have to be some sort of signing. Either way, it's not going to happen for a while.

I guess it makes sense to raise this as a js-sdk event anyway.

File: src/components/views/rooms/MemberDeviceInfo.js (hunk 2 above)
Comment (richvdh) [inline src/components/views/rooms/MemberDeviceInfo.js]: Probably. later.

**Comment by richvdh** [comment]
@dbkr: ptal?

**Comment by dbkr** [comment]
That should do it :) lgtm

**Pull request merged**